### ISSUE TITLE
Dependencies which have transitive=false in ivy file are not included in alignment

### ISSUE THREAD

**Issue by chali** [nebula-plugins/gradle-nebula-integration#53]
I have a dependency which has transitive dependencies defined in ivy file. All of them are members of the same group for alignment. However, one of them is defined with transitive=false.
`<dependency org="com.netflix.governator" name="governator-jetty" rev="1.14.4" conf="compile-&gt;default" transitive="false"/>`

This dependency is not added to a platform for alignment.
`./gradlew dI --dependency governator --configuration compile` gives the following output.

```
> Task :dependencyInsight
com.netflix.governator:governator-api:1.17.5
   variant "runtime" [
      org.gradle.status             = release (not requested)
      org.gradle.usage              = java-runtime (not requested)
      org.gradle.component.category = library (not requested)
   ]
   Selection reasons:
      - By constraint : belongs to platform aligned-platform:governator:1.17.5

com.netflix.governator:governator-api:1.17.5
\--- com.netflix.governator:governator-core:1.17.5
     +--- netflix:library:1.173
     |    \--- compile
     +--- com.netflix.governator:governator-jersey:1.17.5
     |    +--- netflix:library:1.173 (*)
     |    \--- compile
     +--- com.netflix.governator:governator-servlet:1.17.5
     |    +--- netflix:library:1.173 (*)
     |    \--- compile
     \--- compile

com.netflix.governator:governator-core:1.17.5
   variant "runtime" [
      org.gradle.status             = release (not requested)
      org.gradle.usage              = java-runtime (not requested)
      org.gradle.component.category = library (not requested)
   ]
   Selection reasons:
      - Forced
      - By constraint : belongs to platform aligned-platform:governator:1.17.5

com.netflix.governator:governator-core:1.17.5
+--- com.netflix.governator:governator-jersey:1.17.5
|    +--- netflix:library:1.173
|    |    \--- compile
|    \--- compile
\--- com.netflix.governator:governator-servlet:1.17.5
     +--- netflix:library:1.173 (*)
     \--- compile

com.netflix.governator:governator-core:1.14.4 -> 1.17.5
\--- netflix:library:1.173
     \--- compile

com.netflix.governator:governator-jersey:1.17.5
   variant "runtime" [
      org.gradle.status             = release (not requested)
      org.gradle.usage              = java-runtime (not requested)
      org.gradle.component.category = library (not requested)
   ]
   Selection reasons:
      - By constraint : belongs to platform aligned-platform:governator:1.17.5
      - Forced

com.netflix.governator:governator-jersey:1.14.4 -> 1.17.5
\--- netflix:library:1.173
     \--- compile

com.netflix.governator:governator-jetty:1.14.4
   variant "runtime" [
      org.gradle.status             = release (not requested)
      org.gradle.usage              = java-runtime (not requested)
      org.gradle.component.category = library (not requested)
   ]

com.netflix.governator:governator-jetty:1.14.4
\--- netflix:library:1.173
     \--- compile

com.netflix.governator:governator-providers:1.17.5
   variant "runtime" [
      org.gradle.status             = release (not requested)
      org.gradle.usage              = java-runtime (not requested)
      org.gradle.component.category = library (not requested)
   ]
   Selection reasons:
      - By constraint : belongs to platform aligned-platform:governator:1.17.5

com.netflix.governator:governator-providers:1.17.5
\--- com.netflix.governator:governator-jersey:1.17.5
     +--- netflix:library:1.173
     |    \--- compile
     \--- compile

com.netflix.governator:governator-servlet:1.17.5
   variant "runtime" [
      org.gradle.status             = release (not requested)
      org.gradle.usage              = java-runtime (not requested)
      org.gradle.component.category = library (not requested)
   ]
   Selection reasons:
      - By constraint : belongs to platform aligned-platform:governator:1.17.5
      - Forced

com.netflix.governator:governator-servlet:1.14.4 -> 1.17.5
\--- netflix:library:1.173
     \--- compile
```

See `com.netflix.governator:governator-jetty:1.14.4`

Here is the full example project https://github.com/nebula-plugins/gradle-nebula-integration/tree/master/transitive-false-not-aligned

**Comment by ljacomet** [comment]
I know exactly what the issue is in this case. We process `belongsTo` declarations after processing transitive dependencies ... and thus we never considered the case where a dependency was marked non transitive but had a `belongsTo` declared.
Will file an issue and work on a fix!

**Comment by ljacomet** [comment]
Filed gradle/gradle#7620 and fix has been added to PR gradle/gradle#7590

**Comment by ljacomet** [comment]
The fix has been merged and will be in Gradle 5.0 RC2

**Comment by chali** [comment]
Confirmed fix with Gradle 5.0-rc-3
